charts/partials/templates/_network-validator.tpl

```diff
@@ -3,8 +3,11 @@ name: linkerd-network-validator
 image: {{.Values.proxy.image.name}}:{{.Values.proxy.image.version | default .Values.linkerdVersion }}
 imagePullPolicy: {{.Values.proxy.image.pullPolicy | default .Values.imagePullPolicy}}
 {{ include "partials.resources" .Values.proxy.resources }}
-{{- if or .Values.networkValidator.enableSecurityContext }}
+{{- if ne .Values.networkValidator.enableSecurityContext false }}
```

Comment:
tioli but this is generally safest to avoid weird values.yamls that may miss the parent structure

```suggestion
{{- if ne (.Values.networkValidator).enableSecurityContext false }}
```